jetty-ee9/jetty-ee9-annotations/src/main/java/org/eclipse/jetty/ee9/annotations/AnnotationParser.java

```diff
@@ -646,27 +649,60 @@ protected void parseJar(Set<? extends Handler> handlers, Resource jarResource) t
      * @param containingResource the dir or jar that the class is contained within, can be null if not known
      * @param classFile the class file to parse
      * @throws IOException if unable to parse
+     * @deprecated use {@link #parseClass(Set, Resource, Resource)} instead (which uses {@link Resource} instead of {@link Path})
      */
+    @Deprecated(since = "12.0.16", forRemoval = true)
     protected void parseClass(Set<? extends Handler> handlers, Resource containingResource, Path classFile) throws IOException
     {
-        if (LOG.isDebugEnabled())
-            LOG.debug("Parse class from {}", classFile.toUri());
+        try (InputStream inputStream = Files.newInputStream(classFile))
```

Comment:
Null protection?

Reply:
Sure, but it's not possible to reach this point with a null `classFile` now.
Both due to the changes in parse() and the fact that this specific method isn't called anymore by anything in Jetty.